### DISCUSSION TITLE
Named export from prime-react not found during build

### DISCUSSION THREAD

**Discussion by snake-py** [vikejs/vike#1023] [Discussion] [answered]
Hello,

I am not sure if this is a [primereact](https://primereact.org/) issue. But it works if I import this in other projects, where I don't use VPS. Some components from primereact cannot be imported during the build, also in the error message I am seeing a different import stmt. 

For instance :

This import works during dev and also during build in project without VPS 
```ts
import { ProgressSpinner } from 'primereact/progressspinner';
```

In VPS project during build I get:

```
vite-plugin-ssr v0.4.132 pre-rendering HTML...
Vite uses plugin set for app
file:///home/fabio/Desktop/workdigital/portal/frontend2.0/dist/server/entries/src_pages_login_login-page.mjs:5
import { ProgressSpinner } from "primereact/progressspinner/progressspinner.esm.js";
         ^^^^^^^^^^^^^^^
SyntaxError: Named export 'ProgressSpinner' not found. The requested module 'primereact/progressspinner/progressspinner.esm.js' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from 'primereact/progressspinner/progressspinner.esm.js';
const { ProgressSpinner } = pkg;

    at ModuleJob._instantiate (node:internal/modules/esm/module_job:128:21)
    at ModuleJob.run (node:internal/modules/esm/module_job:194:5)
    at async Promise.all (index 0)
    at ESMLoader.import (node:internal/modules/esm/loader:385:24)
    at Object.pageFile.loadFile (/home/fabio/Desktop/workdigital/portal/frontend2.0/node_modules/vite-plugin-ssr/dist/cjs/shared/getPageFiles/parseGlobResults.js:35:40)
    at async Promise.all (index 0)
    at loadPageFilesServerSide (/home/fabio/Desktop/workdigital/portal/frontend2.0/node_modules/vite-plugin-ssr/dist/cjs/shared/getPageFiles/analyzePageServerSide/loadPageFilesServerSide.js:10:5)
    at async Promise.all (index 0)
    at loadPageFilesServer (/home/fabio/Desktop/workdigital/portal/frontend2.0/node_modules/vite-plugin-ssr/dist/cjs/node/runtime/renderPage/loadPageFilesServer.js:15:110)
    at /home/fabio/Desktop/workdigital/portal/frontend2.0/node_modules/vite-plugin-ssr/dist/cjs/node/prerender/runPrerender.js:246:48
```

Any idea why this could happen?

I am prerendering in case this makes a difference:

```ts
      ssr({
        prerender: true
      })
```

**Comment by brillout** [comment]
Can you create a reproduction? Working around these issues is on the roadmap, and the more reproductions I can get the better.

To answer your question: https://github.com/cyco130/vite-plugin-cjs-interop but there may be a simpler solution which I'll try once I get my hands on a reproduction.

**Reply by snake-py** [reply]
@brillout I tried to make a reproducable error, the build fails if I set prerender to true.

Here is the repository. https://github.com/snake-py/vps-cjs-issue

**Reply by brillout** [reply]
https://github.com/brillout/vps-cjs-issue/commit/9dc650bbddeac3ef78ece36596fc30f478fa238e

**Reply by brillout** [reply]
Let me know whether it works for other imports as well.

**Reply by snake-py** [reply]
@brillout what will ssr no external do will it then ignore all components for ssr from primereact?

**Reply by brillout** [reply]
https://vitejs.dev/guide/ssr.html#ssr-externals

**Reply by brillout** [reply, marked as answer]
https://vite-plugin-ssr.com/primereact

> PrimeReact is wrongly published (see https://github.com/primefaces/primereact/issues/4675).
> 
> This example uses [this workaround](https://github.com/brillout/vps-primereact/commit/9dc650bbddeac3ef78ece36596fc30f478fa238e) to make it work until PrimeReact fixes the issue. (For more information see discussion at https://github.com/brillout/vite-plugin-ssr/discussions/1023.)
